app/javascript/packs/admin_evaluations/new_evaluation.jsx

```diff
@@ -0,0 +1,220 @@
+    this.handleMarkChange = this.handleMarkChange.bind(this);
+    this.handleAddGoal = this.handleAddGoal.bind(this);
+    this.handleRemoveGoal = this.handleRemoveGoal.bind(this);
+  }
+
+  componentWillMount() {
+    this.state.evaluation.goals_attributes[0]._destroy = true;
+  }
+
+  componentDidMount() {
+    $("#add-goal").hide();
+    $("#submit").hide();
+  }
+
+  handleSubmit() {
+    const dueDate = this.state.evaluation.dueDate;
+    const evaluationGoals = this.state.evaluation.goals_attributes;
+    Fetch.postJSON(`/api/v1/admin/user/users/${this.userId}/evaluations`, {
+      evaluation: {
+        user_id: this.userId,
+        due_date: dueDate,
+        goals_attributes: evaluationGoals,
+      },
+    })
+      .then(() => {
+        this.msg.success('Successfully created evaluation!');
+        this.setState({ redirect: true });
+      }).catch((errorResponse) => {
+      this.msg.error(errorResponse.errors);
+    });
+  }
+
+  handleDueDateChange(e) {
+    let evaluation = this.state.evaluation;
+    evaluation.dueDate = e.target.value;
+    this.setState({ evaluation: this.state.evaluation });
+  }
+
+  handleGoalNameChange(e) {
+    let goal = this.emptyGoal;
+    goal.name = e.target.value;
+    this.setState({ goalName: e.target.value});
+    if (goal.name.length > 0) {
+      $("#add-goal").show();
+    } else {
+      $("#add-goal").hide();
+    }
+  }
+
+  handleMarkChange(e) {
+    let goal = this.emptyGoal;
+    goal.mark = parseInt(e.target.value);
+    this.setState({ goalMark: e.target.value});
+  }
+
+  handleAddGoal() {
+    this
+      .state
+      .evaluation
+      .goals_attributes
+      .push(Object.assign({}, this.emptyGoal));
+    this.setState({ evaluation: this.state.evaluation });
+    this.setState({ goalName: "" });
+    this.setState({ goalMark: "" });
+  }
+
+  handleRemoveGoal(goal) {
+    goal._destroy = true;
+    this.setState({ evaluation: this.state.evaluation });
+  }
+
+  render() {
+    if (this.state.redirect) {
+      return (
+        <Redirect to={'/admin/users/' + this.userId + '/evaluations'} />
+      );
+    }
+
+    const submitButton = this.state.evaluation.goals_attributes.length > 1 && this.state.evaluation.dueDate !== '' ?
+      (
+        <button
+          id="submit"
+          className="btn btn-success"
+          onClick={this.handleSubmit}
+        >
+          Create Evaluation
+        </button>
+      )
+      :
+      null;
+
+    const goals = this.state.evaluation.goals_attributes.map((goal, index) => {
+      if (goal._destroy === false) {
+        return (
+          <tr key={index}>
+            <td>{goal.name}</td>
+            <td>{goal.mark}</td>
+            <td>
+              <button
+                className="btn btn-danger"
+                  onClick={e => this.handleRemoveGoal(goal)}>
+                Delete
+              </button>
+            </td>
+          </tr>
+        );
+      } else {
```

Comment:
If an `if` block contains a `return` statement, the `else` block becomes unnecessary